Add own waypoint fails when geocache has already >100 waypoints defined

From german geocaching forum:
https://geoclub.de/forum/viewtopic.php?f=155&t=67870&start=700#p1276867

- open https://coord.info/GC6HH0B 
- goto waypoints
- add an own wapoint
- c:geo crashes

Maybe related to #6243 

I could reproduce the bug, but couldn't yet look into the logs what happens.

Looks like we don't support more than 100 waypoints:

```
Caused by: java.lang.IllegalStateException: too many waypoints, unable to assign unique prefix
  at cgeo.geocaching.models.Geocache.assignUniquePrefix(Geocache.java:1409)
  at cgeo.geocaching.models.Geocache.addOrChangeWaypoint(Geocache.java:1366)
  at cgeo.geocaching.EditWaypointActivity$1SaveWptTask.doInBackground(EditWaypointActivity.java:659)
  at cgeo.geocaching.EditWaypointActivity$1SaveWptTask.doInBackground(EditWaypointActivity.java:637)
  at android.os.AsyncTask$2.call(AsyncTask.java:305)
  at java.util.concurrent.FutureTask.run(FutureTask.java:237)
  at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243) 
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
  at java.lang.Thread.run(Thread.java:761) 
```

--- a/main/src/cgeo/geocaching/models/Geocache.java
+++ b/main/src/cgeo/geocaching/models/Geocache.java
@@ -88,7 +88,8 @@ import org.apache.commons.lang3.StringUtils;
  */
 public class Geocache implements IWaypoint {
 
-    private static final int OWN_WP_PREFIX_OFFSET = 17;
+    private static final String WP_PREFIX_CHARS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
     private long updated = 0;
     private long detailedUpdate = 0;
     private long visitedDate = 0;
@@ -1388,7 +1389,7 @@ public class Geocache implements IWaypoint {
     }
 
     /*
-     * Assigns a unique two-digit (compatibility with gc.com)
+     * Assigns a unique two-character (compatibility with gc.com)
      * prefix within the scope of this cache.
      */
     private void assignUniquePrefix(final Waypoint waypoint) {
@@ -1398,8 +1399,9 @@ public class Geocache implements IWaypoint {
             assignedPrefixes.add(wp.getPrefix());
         }
 
-        for (int i = OWN_WP_PREFIX_OFFSET; i < 100; i++) {
-            final String prefixCandidate = String.valueOf(i);
+        final int length = WP_PREFIX_CHARS.length();
+        for (int i = 0; i < length * length; i++) {
+            final String prefixCandidate = "" + WP_PREFIX_CHARS.charAt(i / length) + WP_PREFIX_CHARS.charAt(i % length);
             if (!assignedPrefixes.contains(prefixCandidate)) {
                 waypoint.setPrefix(prefixCandidate);
                 return;

